setup.py

```diff
@@ -29,6 +29,12 @@
 if os.getenv("SNOWFLAKE_IS_PYTHON_RUNTIME_TEST", False):
     REQUIRED_PYTHON_VERSION = ">=3.8"
 
+PANDAS_REQUIREMENTS = (
+    [
+        f"snowflake-connector-python[pandas]{CONNECTOR_DEPENDENCY_VERSION}",
```

Comment:
It is weird to me that we require pandas through a transient dependency on the connector. Is that right?

Reply:
snowflake-snowpark-python[modin] gets pandas itself via the modin dependency as well. `snowflake-snowpark-python[pandas]` installs snowflake-connector-python together with the dependencies that the connector [needs](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-pandas#installation) to interoperate with pandas. Those [seem to be] both (https://github.com/snowflakedb/snowflake-connector-python/blob/eca86b9fcd5da9da722d18edc756c90ae940304b/setup.cfg#L95-L97) `pandas>=1.0.0,<3.0.0` and `pyarrow>=13.0.0`.